src/Features/Core/Portable/DesignerAttribute/DesignerAttributeDiscoveryService.cs

```diff
@@ -156,20 +165,6 @@ private static async ValueTask<bool> HasDesignerCategoryTypeAsync(Project projec
             // change if anything it depends on changes).
             var lazyProjectVersion = AsyncLazy.Create(project.GetSemanticVersionAsync);
 
-            // Switch to frozen semantics if requested.  We don't need to wait on generators to run here as we want to
-            // be lightweight.  We'll also continue running in the future.  So if any changes to happen that are
-            // important to pickup, then we'll see it in the future.  But this avoids constant churn here trying to do
-            // things like building skeletons.
-            if (useFrozenSnapshots)
-            {
-                var document = specificDocument ?? project.Documents.FirstOrDefault();
-                if (document is null)
-                    return;
-
-                project = document.WithFrozenPartialSemantics(cancellationToken).Project;
-                specificDocument = specificDocument is null ? null : project.GetRequiredDocument(specificDocument.Id);
-            }
```

Comment:
>} [](http://example.com/codeflow?start=12&length=1)

nit: the "if (specificDocument != null)" check below seems like it should be guarding the other ScanForDesignerCategoryUsageAsync call, as it's the one that uses specificDocument.

Reply:
i'll rearrange to make clearer.